gemfirexd/tools/src/test/java/com/pivotal/gemfirexd/jdbc/transactions/snapshot/SnapshotTransactionTest.java

```diff
@@ -100,6 +94,120 @@ public void testRVVSnapshotContains() throws Exception {
 
   }
 
+  public void testAutoCommitWithRCConflict() throws Exception {
+    Connection conn= getConnection();
+    Statement st = conn.createStatement();
+    st.execute("Create table tran.t1 (c1 int not null , c2 int not null, primary key(c1)) replicate"+getSuffix());
+    conn.commit();
+    conn.setTransactionIsolation(Connection.TRANSACTION_NONE);
+    conn.setAutoCommit(true);
+    final TXManagerImpl txMgrImpl = (TXManagerImpl)Misc.getGemFireCache()
+        .getCacheTransactionManager();
+    // Region r= cache.getRegion("APP/T1");
+    final Region<Object, Object> r = Misc.getRegionForTable("TRAN.T1", true);
+    final Object key = getGemFireKey(10, r);
+    this.gotConflict = false;
+    Thread thread = new Thread(new Runnable() {
+
+      @Override
+      public void run() {
+        assertNotNull(r);
+        txMgrImpl.begin(IsolationLevel.SNAPSHOT, null);
+        r.put(key, new SQLInteger(20)); // create a conflict here.
+        TXStateInterface txi = txMgrImpl.internalSuspend();
+        assertNotNull(txi);
+        txMgrImpl.resume(txi);
+        txMgrImpl.commit();
+      }
+    });
+    thread.start();
+    thread.join();
+
+
+    st.execute("insert into tran.t1 values (10, 20)");
+
+    //st.execute("insert into t1 values (20, 20)");
+
+    //Misc.getGemFireCache().getCacheTransactionManager().begin(IsolationLevel.SNAPSHOT, null);
+    ResultSet rs = st.executeQuery("Select * from tran.t1 where c2 = 10");
+    int numRows = 0;
+    while (rs.next()) {
+      numRows++;
+    }
+    assertEquals("ResultSet should contain two rows ", 1, numRows);
+
+    rs.close();
+   // conn.setTransactionIsolation(Connection.TRANSACTION_NONE);
+    //Misc.getGemFireCache().getCacheTransactionManager().commit();
+    conn.commit();
+
+    // Close connection, resultset etc...
+
+    st.close();
+    conn.close();
+  }
+
+
+  public void testAutoCommitWithIndexRC() throws Exception {
+    Connection conn= getConnection();
+    Statement st = conn.createStatement();
+    st.execute("Create table t1 (c1 int not null , c2 int not null, primary key(c1)) replicate"+getSuffix());
+    conn.commit();
+    conn.setTransactionIsolation(Connection.TRANSACTION_NONE);
+    conn.setAutoCommit(true);
+
+    st.execute("insert into t1 values (10, 10)");
+
+    //st.execute("insert into t1 values (20, 20)");
+
+    Misc.getGemFireCache().getCacheTransactionManager().begin(IsolationLevel.SNAPSHOT, null);
+    ResultSet rs = st.executeQuery("Select * from t1 where t1.c2 = 10");
+    int numRows = 0;
+    while (rs.next()) {
+      numRows++;
+    }
+    assertEquals("ResultSet should contain two rows ", 1, numRows);
```

Comment:
Again instead of two in the message you must be meaning one. BTW I am missing something basic. The name of the test has RC in it and I don't see that level ever being set anywhere.